### ISSUE TITLE
Calculations for new state start at non-zero values

### ISSUE THREAD

**Issue by raintonr** [iobroker-community-adapters/ioBroker.sourceanalytix#259]
I created a test state under Javacript adapter and made sure it's value was set to zero. Then turned on sourceanalytix calculations for that state.

Then created this script and ran *one* time...

```javascript
const state = 'sourceanalytixTest.watts';

// 1500w for 30s is 0.0125kWh
log(`Turning on: ${state}`, 'info');
setState(state, 1500, true);
setTimeout(() => {
    log(`Turning off: ${state}`, 'info');
    setState(state, 0, true);
}, 30000);
```

... which logged this output, showing the test power state was indeed at 1500W for precisely 30 seconds...

```
14:37:31.820 | info | javascript.0 (23098) Start javascript script.js.Tests.sourceanalytixTest
14:37:31.978 | info | javascript.0 (23098) script.js.Tests.sourceanalytixTest: Turning on: sourceanalytixTest.watts
14:38:01.981 | info | javascript.0 (23098) script.js.Tests.sourceanalytixTest: Turning off: sourceanalytixTest.watts
```

... yet this lead to these results:

![image](https://user-images.githubusercontent.com/3794098/100763529-8d909900-33f5-11eb-9b63-19584f694744.png)

As you can see, instead of recording energy usage of 0.0125kWh, the value is that (rounded) plus 1.5kWh. Clearly an error.

I find it suspicious that 1.5kWh was added for a power input of 1.5kW for 30 seconds.

FWIW, I ran the script above a further 2 times. The energy calculation should be 0.0375kWh after that. Even account for the 1.5kWh initialisation discrepancy it should be 1.5375 but in fact the value shown by sourceanalytix was 1.541. Clearly still needs work :(

**Comment by raintonr** [comment]
Having a quick look at the code, it's probably to do with this line...

https://github.com/iobroker-community-adapters/ioBroker.sourceanalytix/blob/master/main.js#L1243

... which is reached when no prior timestamp or value are known. In this case it's illogical to return anything but a zero value.

I notice this bug was introduced in https://github.com/iobroker-community-adapters/ioBroker.sourceanalytix/commit/4a9a63e39f507fb17c69bcb89ff69dcfb2154bd2. Prior to that it would have worked as I expect. What was this commit supposed to achieve?

**Comment by raintonr** [comment]
> Having a quick look at the code, it's probably to do with this line...

I think that's also what is causing the calculated value to be too large also. Check out this debug log:

```
2020-12-01 17:56:41.020  - info: javascript.0 (23098) Start javascript script.js.Tests.sourceanalytixTest_js
2020-12-01 17:56:41.108  - info: javascript.0 (23098) script.js.Tests.sourceanalytixTest_js: Turning on: sourceanalytixTest.watts
2020-12-01 17:56:41.110  - info: javascript.0 (23098) script.js.Tests.sourceanalytixTest_js: registered 0 subscriptions and 0 schedules
2020-12-01 17:56:41.130  - debug: sourceanalytix.0 (25190) state javascript.0.sourceanalytixTest.watts changed : {"val":1500,"ack":true,"ts":1606841801112,"q":0,"from":"system.adapter.javascript.0","user":"system.user.admin","lc":1606841801112} SourceAnalytix calculation executed
2020-12-01 17:56:41.132  - debug: sourceanalytix.0 (25190) Calculation for javascript.0.sourceanalytixTest.watts with values : {"val":1500,"ack":true,"ts":1606841801112,"q":0,"from":"system.adapter.javascript.0","user":"system.user.admin","lc":1606841801112} and configuration : {"stateDetails":{"alias":"","consumption":false,"costs":false,"deviceName":"javascript__0__sourceanalytixTest__watts","financialCategory":"costs","headCategory":"consumed","meter_values":false,"name":"watts","stateType":"Electricity","stateUnit":"W","useUnit":"kWh"},"calcValues":{"currentValue":1.5495480000000001,"start_day":0,"start_month":0,"start_quarter":0,"start_week":0,"start_year":0,"valueAtDeviceReset":0,"previousReadingWatt":0,"previousReadingWattTs":1606838541681},"prices":{"basicPrice":"0","unitPrice":"0.14184"}}
2020-12-01 17:56:41.134  - debug: sourceanalytix.0 (25190) Watt to kWh for javascript.0.sourceanalytixTest.watts current reading : 1500 previousReading : {"stateDetails":{"alias":"","consumption":false,"costs":false,"deviceName":"javascript__0__sourceanalytixTest__watts","financialCategory":"costs","headCategory":"consumed","meter_values":false,"name":"watts","stateType":"Electricity","stateUnit":"W","useUnit":"kWh"},"calcValues":{"currentValue":1.5495480000000001,"start_day":0,"start_month":0,"start_quarter":0,"start_week":0,"start_year":0,"valueAtDeviceReset":0,"previousReadingWatt":0,"previousReadingWattTs":1606838541681},"prices":{"basicPrice":"0","unitPrice":"0.14184"}}
2020-12-01 17:56:41.135  - debug: sourceanalytix.0 (25190) Watt to kWh outcome for javascript.0.sourceanalytixTest.watts : {"currentValue":1.5495480000000001,"start_day":0,"start_month":0,"start_quarter":0,"start_week":0,"start_year":0,"valueAtDeviceReset":0,"previousReadingWatt":1500,"previousReadingWattTs":1606841801112}
2020-12-01 17:56:41.136  - debug: sourceanalytix.0 (25190) Recalculated value 1.5510975480000002
2020-12-01 17:56:41.137  - debug: sourceanalytix.0 (25190) New reading 1.5510975480000002 bigger than stored value 1.5495480000000001 processing normally
2020-12-01 17:56:41.137  - debug: sourceanalytix.0 (25190) Set calculated value 1.5510975480000002 on state : javascript__0__sourceanalytixTest__watts.Current_Reading}
2020-12-01 17:56:41.138  - debug: sourceanalytix.0 (25190) ActiveStatesArray {"currentValue":1.5510975480000002,"start_day":0,"start_month":0,"start_quarter":0,"start_week":0,"start_year":0,"valueAtDeviceReset":0,"previousReadingWatt":1500,"previousReadingWattTs":1606841801112})
2020-12-01 17:56:41.139  - debug: sourceanalytix.0 (25190) roundDigits with 1.5510975480000002 rounded 1.551
```

Note my test script starts and changes it's power state from 0W to 1500W. This should have no effect on the kWh calculation as it doesn't matter how long it has been since that zero value was set... zero multiplied by anything is still zero. However, you note that method in fact returns a non-zero value which is added to the meter reading which increments (1.5495480000000001 -> 1.5510975480000002).

Again, this due to https://github.com/iobroker-community-adapters/ioBroker.sourceanalytix/commit/4a9a63e39f507fb17c69bcb89ff69dcfb2154bd2.

**Comment by DutchmanNL** [comment]
Solved in 0.4.8-Beta.0 which will be released this weekend, alpha 0.4.8-Alpha.13 is already available for install and has all fixes included.
Please do not install any version of 0.4.8 lower than alpha 13 ! (Or wait for beta release on Sunday 15-01-2021)

thanks for your help !
